src/Connection.php

```diff
@@ -154,77 +149,72 @@ public static function connect($dsn, $user = null, $password = null, $args = [])
-                ], $args));
-
-                break;
-            case 'sqlite':
-                $c = new static(array_merge([
-                    'connection' => static::getPDO($dsn),
-                    'query_class' => Sqlite\Query::class,
-                    'driverType' => $dsn['driverType'],
-                ], $args));
-
-                break;
-            case 'oci':
-                $c = new Oracle\Connection(array_merge([
-                    'connection' => static::getPDO($dsn),
-                    'driverType' => $dsn['driverType'],
-                ], $args));
-
-                break;
-            case 'oci12':
-                $dsn['dsn'] = str_replace('oci12:', 'oci:', $dsn['dsn']);
-                $c = new Oracle\Version12\Connection(array_merge([
-                    'connection' => static::getPDO($dsn),
-                    'driverType' => $dsn['driverType'],
-                ], $args));
-
-                break;
-            case 'pgsql':
-                $c = new Postgresql\Connection(array_merge([
-                    'connection' => static::getPDO($dsn),
-                    'driverType' => $dsn['driverType'],
-                ], $args));
-
-                break;
-            case 'dumper': // deprecated - will be removed in dec-2020
-            case 'stopwatch':
-                $c = new Debug\Stopwatch\Connection(array_merge([
-                    'connection' => static::connect($dsn['rest'], $dsn['user'], $dsn['pass']),
-                ], $args));
-
-                break;
-            case 'counter': // deprecated - will be removed in dec-2020
-            case 'profile':
-                $c = new Debug\Profiler\Connection(array_merge([
-                    'connection' => static::connect($dsn['rest'], $dsn['user'], $dsn['pass']),
-                ], $args));
-
-                break;
-            // let PDO handle the rest
-            default:
-                $c = new static(array_merge([
-                    'connection' => static::connect(static::getPDO($dsn)),
-                ], $args));
+        /**
+         * @var Connection $connectionClass
+         */
+        $connectionClass = self::resolveConnection($dsn['driverType']);
+
+        return new $connectionClass(array_merge([
+            'connection' => $connectionClass::establishConnection($dsn),
+        ], $args));
+    }
+
+    /**
+     * Adds connection class to the registry for resolving in Connection::resolve method.
+     *
+     * Can be used as:
+     *
+     * Connection::register('mysql', MySQL\Connection::class), or
+     * MySQL\Connection::register()
+     *
+     * CustomDriver\Connection must be descendant of Connection class.
+     *
+     * @param string $connectionType
+     * @param string $connectionClass
+     */
+    public static function registerConnection($connectionType = null, $connectionClass = null)
+    {
+        if (!$connectionClass && is_a($connectionType, self::class, true)) {
```

Comment:
This is very bad design. We should never swap any arguments. The method has some header with some params order and it should also work that way.

Reply:
changed now

Reply:
![image](https://user-images.githubusercontent.com/2228672/86511969-129f6400-bdfe-11ea-90b0-2514ec2ed66c.png)

but you introduces some more balast!

what is the realworld scenario where this provide any significant value for standard user?

if someone wants to register one class for more types, then it can be simply called twice!!!

Reply:
It is like asking why we need e-mail if we can write letters on paper :)
It is a facilitation - but registering a list of connections removed now to keep things rudimentary.

Reply:
rudimentary is perfect! :)